profiler/src/ProfilerEngine/Datadog.Profiler.Native/PInvoke.cpp

```diff
@@ -109,4 +87,36 @@ extern "C" void __stdcall SetApplicationInfoForAppDomain(const char* runtimeId,
         serviceName ? serviceName : std::string(),
         environment ? environment : std::string(),
         version ? version : std::string());
+}
+
+extern "C" void __stdcall SetEndpointForTrace(const char* runtimeId, uint64_t traceId, const char* endpoint)
+{
+    if (!CorProfilerCallback::GetClrLifetime()->IsRunning())
+    {
+        return;
+    }
+
+    const auto profiler = CorProfilerCallback::GetInstance();
+
+    if (profiler == nullptr)
+    {
+        Log::Error("SetEndpointForTrace is called BEFORE CLR initialize");
+        return;
+    }
+
+    if (runtimeId == nullptr)
+    {
+        Log::Error("SetEndpointForTrace was called with an empty runtime id");
+        return;
+    }
+
+    if (endpoint == nullptr)
+    {
+        // It could happen that the endpoint is empty, but the tracer should check before making the call,
+        // to avoid the cost of the p/invoke
+        Log::Warn("SetEndpointForTrace was called with an empty endpoint");
+        return;
+    }
```

Comment:
Just to confirm, we don't expect these to ever actually happen, right? (just thinking about log levels)

Reply:
Yeah, but that reminds me that I forgot to check if `ResourceName` is null in the tracer part 😬

Reply:
nvm, the check is in `ContextTracker.SetEndpoint`